### DISCUSSION TITLE
CTkColorPicker

### DISCUSSION THREAD

**Discussion by Akascape** [TomSchimansky/CustomTkinter#929] [Showcase]
**I made a color picker for customtkinter!**
## [**CTkColorPicker**](https://github.com/Akascape/CTkColorPicker)

![Screenshot](https://user-images.githubusercontent.com/89206401/209182773-d76bf05c-610e-4297-aec5-7bb61a11d6d3.jpg)

**Hope it will help :)**

**Comment by avalon60** [comment]
Akash, I might find this useful. I was considering writing something myself. 
Presumably the slider graduates the tint of a chosen colour (lighter/darker)? 
Is it limited to Windows?

**Reply by Akascape** [reply]
@avalon60 
> Presumably the slider graduates the tint of a chosen colour (lighter/darker)?

The slider can be used to get darker/lighter shades of the chosen color.

> Is it limited to Windows?

No, it is also tested on mac and linux, works the same everywhere. 

![](https://user-images.githubusercontent.com/89206401/227775283-d672cfd6-a3ff-4969-83e8-4d01c61b61ec.png)

**Reply by avalon60** [reply]
Cool - I think I'll give it a whirl!

**Reply by avalon60** [reply]
Akash, is it possible to set the initial colour?

**Reply by Akascape** [reply]
@avalon60 we can pass the initial color with `initial_color="#hex_string"`, but this is in beta stage, may not work for all colors.

**Reply by JPLost** [reply]
Just found this nice addition. Is there a chance to define which way the color code and slider are positioned (horizontal or vertical)?
![kuva](https://github.com/TomSchimansky/CustomTkinter/assets/48565494/eb2e3bc6-05ef-45cf-9df7-5c168b7e8318)

**Reply by Akascape** [reply]
@JPLost orientation option is not implemented because nobody asked for it.
But now I will add the horizontal orientation too.

**Reply by Akascape** [reply]
@JPLost I just implemented this option, now you can pass `orientation="horizontal"` as parameter. First update the package.

**Comment by Akascape** [comment]
Now you can also add the color picker as a widget, update the package and see details.

![Screenshot](https://github.com/Akascape/CTkColorPicker/assets/89206401/ca03751a-90a3-45e6-8ba4-745a92ab1c12)

Another color picker (scale) is available in this package: [tkmacosx](https://github.com/Saadmairaj/tkmacosx)

<img src="https://user-images.githubusercontent.com/46227224/114269070-55900700-9a22-11eb-8de5-38885fad5cf8.gif">

Another classic color picker is available here:
[TkColorPicker](https://github.com/j4321/tkColorPicker)

<img src="https://user-images.githubusercontent.com/89206401/242483953-1baf0d80-e892-4745-ae26-153a0382a3bc.jpg">

**Comment by morpheuscodes** [comment]
Oh shit Well done. This is huge for any application with color-related activities.
